qiskit/providers/fake_provider/utils/backend_converter.py

```diff
@@ -113,17 +113,19 @@ def convert_to_target(conf_dict: dict, props_dict: dict = None, defs_dict: dict
         inst_map = pulse_defs.instruction_schedule_map
         for inst in inst_map.instructions:
             for qarg in inst_map.qubits_with_instruction(inst):
-                sched = inst_map.get(inst, qarg)
+                try:
+                    qargs = tuple(qarg)
+                except TypeError:
+                    qargs = (qarg,)
+                # Do NOT call .get method. This parses Qpbj immediately.
+                # This operation is computationally expensive and should be bypassed.
+                calibration_entry = inst_map._map[inst][qargs]
```

Comment:
Sounds like a good idea. Done in [add API for getting calibration entry](https://github.com/Qiskit/qiskit-terra/pull/8885/commits/41217fdfaf087579e9d92efc16d7dde654f81e34). I added new API as a protected method since this is not expected to interface with users. During this change I also noticed backend converter exists for different places, namely `qiskit.providers.fake_provider.utils.backend_converter.convert_to_target` and `qiskit.providers.backend_compat.convert_to_target`. Do you know why the similar code exists in different places.

Reply:
I think because in #8759 what I wrote for that PR was more general and I intended to follow up and remove the fake provider version which was more targeted to ibm backends but forgot to. I can push a PR to do this now

Reply:
Oh, I guess one key difference between the implementations is the fake provider version works with a raw dictionary without creating an intermediate `BackendConfiguration` and `BackendProperties`  object. IIRC, we did that for efficiency because we can just parse the raw json files and go straight to a target. I'm not sure if it's worth deleting the second copy or not (or how much of a difference that makes in practice). Also at some far future date when `BackendConfiguration` and `BackendProperties` are deprecated and then removed we will want the dictionary version too.